From 5bc2ee6d9dbaa49ec1b80fcb4537d02c0eff2c8a Mon Sep 17 00:00:00 2001
From: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
Date: Wed, 10 May 2023 19:59:39 +0530
Subject: [PATCH] wifi: ath12k: Handle the vdev id mismatch HTT Tx complete

Unhandled HTT Tx completion status skb are not freed in Tx completion
procedure. When vdev id mismatch HTT Tx status recieved, we leak the
skb memory due to unhandled situation. Over the period it leads to
memory unavailable situation in the system which end up in crash.

Signed-off-by: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_tx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath12k/dp_tx.c b/drivers/net/wireless/ath/ath12k/dp_tx.c
index 2055b47d0b08..9ae218390027 100644
--- a/drivers/net/wireless/ath/ath12k/dp_tx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_tx.c
@@ -708,6 +708,7 @@ ath12k_dp_tx_process_htt_tx_complete(struct ath12k_base *ab,
 		break;
 	case HAL_WBM_REL_HTT_TX_COMP_STATUS_REINJ:
 	case HAL_WBM_REL_HTT_TX_COMP_STATUS_INSPECT:
+	case HAL_WBM_REL_HTT_TX_COMP_STATUS_VDEVID_MISMATCH:
 		ath12k_dp_tx_free_txbuf(ab, msdu, mac_id, tx_ring, skb_ext_desc);
 		break;
 	case HAL_WBM_REL_HTT_TX_COMP_STATUS_MEC_NOTIFY:
-- 
2.17.1

